controllers/nnf_workflow_controller_container_helpers.go

```diff
@@ -73,6 +73,11 @@ func (c *nnfUserContainer) createMPIJob() error {
 			Namespace: c.workflow.Namespace,
 		},
 	}
+
+	if c.profile.Data.MPISpec == nil {
```

Comment:
This, and the MPIReplicaSpecs checks below, should be moved to the webhook...in fact, the webhook already has this stuff.

Reply:
This is stuff I ran into when attempting to unit test (i.e. no webhook). I suppose this is a belt and suspenders approach. Too much?

Reply:
I would have expected all of that to be caught here: 
https://github.com/NearNodeFlash/nnf-sos/blob/master/api/v1alpha1/nnfcontainerprofile_webhook.go#L51-L80
Does something in the webhook need to be adjusted?

Reply:
How were you unit-testing this without having the webhook running?

Reply:
Well this is fun:

In suite_test.go, comment out the webhook:
https://github.com/NearNodeFlash/nnf-sos/blob/master/controllers/suite_test.go#L184

Then notice that the webhook-present test doesn't fail:
https://github.com/NearNodeFlash/nnf-sos/blob/master/controllers/nnf_workflow_controller_test.go#L1241

Reply:
> How were you unit-testing this without having the webhook running?

I was able to get this method unit tested up until the `Create()` point - where mpi-operator is needed. Seems like I was only able to get that far based on your findings that the webhook won't fail? I'll dig into this more.

Reply:
I've removed the checks in favor of the webhook.